File: src/main/java/com/nickcoblentz/montoya/utils/CopyRequestResponse.java
package com.nickcoblentz.montoya.utils;


import burp.api.montoya.BurpExtension;
import burp.api.montoya.MontoyaApi;
import burp.api.montoya.core.Registration;
import burp.api.montoya.http.message.HttpRequestResponse;
import burp.api.montoya.logging.Logging;
import burp.api.montoya.ui.contextmenu.ContextMenuEvent;
import com.nickcoblentz.montoya.MontoyaLogger;

import javax.swing.*;
import java.awt.*;
import java.awt.datatransfer.Clipboard;
import java.awt.datatransfer.ClipboardOwner;
import java.awt.datatransfer.StringSelection;
import java.awt.datatransfer.Transferable;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public class CopyRequestResponse implements BurpExtension {
    private MontoyaApi _api;
    private Registration _contextItemsProviderRegistration;



    @Override
    public void initialize(MontoyaApi api)
    {
        _api = api;
        MontoyaLogger logger = new MontoyaLogger(api,MontoyaLogger.DebugLogLevel);
        logger.debugLog(this.getClass().getName(),"Plugin Starting...");
        api.extension().setName("Copy Request/Response");
        _contextItemsProviderRegistration = api.userInterface().registerContextMenuItemsProvider(new CopyRequestResponseContextMenuProvider(api));
        logger.debugLog(this.getClass().getName(),"Finished");
/*
        api.logging().logToOutput(api.persistence().preferences().getString("com.nickcoblentz.montoya.explorepreferences.keyname"));
        api.persistence().preferences().stringKeys().forEach(key ->{
            api.logging().logToOutput(key);
        });

 */
    }



}

File: src/main/java/com/nickcoblentz/montoya/utils/CopyRequestResponse.kt
package com.nickcoblentz.montoya.utils

import burp.api.montoya.BurpExtension
import burp.api.montoya.MontoyaApi
import burp.api.montoya.ui.hotkey.HotKeyContext
import com.nickcoblentz.montoya.LogLevel
import com.nickcoblentz.montoya.MontoyaLogger


class CopyRequestResponse : BurpExtension {
    private lateinit var _api: MontoyaApi
    private lateinit var _copyHandler: CopyRequestResponseHandler



    override fun initialize(api: MontoyaApi) {
        _api = api
        val logger = MontoyaLogger(api, LogLevel.DEBUG)
        logger.debugLog(this.javaClass.getName(), "Plugin Starting...")
        api.extension().setName("Copy Request/Response")
        _copyHandler = CopyRequestResponseHandler(_api)
        api.userInterface().registerContextMenuItemsProvider(CopyRequestResponseContextMenuProvider(api,_copyHandler))
        api.userInterface().registerHotKeyHandler(
            HotKeyContext.HTTP_MESSAGE_EDITOR,
            "Ctrl+Shift+C",
            CopyRequestResponseHotKeyProvider(api,true,CopyRequestResponseHandler.CopyMode.RequestFullResponseFull))
        api.userInterface().registerHotKeyHandler(
            HotKeyContext.HTTP_MESSAGE_EDITOR,
            "Ctrl+Alt+C",
            CopyRequestResponseHotKeyProvider(api,true,CopyRequestResponseHandler.CopyMode.RequestFullResponseHeaders))
        /*api.userInterface().registerHotKeyHandler(
            HotKeyContext.HTTP_MESSAGE_EDITOR,
            "",
            CopyRequestResponseHotKeyProvider(api,true,CopyRequestResponseHandler.CopyMode.URLResponseFull))
        api.userInterface().registerHotKeyHandler(
            HotKeyContext.HTTP_MESSAGE_EDITOR,
            "",
            CopyRequestResponseHotKeyProvider(api,true,CopyRequestResponseHandler.CopyMode.URLResponseHeaders))
        api.userInterface().registerHotKeyHandler(
            HotKeyContext.HTTP_MESSAGE_EDITOR,
            "",
            CopyRequestResponseHotKeyProvider(api,false,CopyRequestResponseHandler.CopyMode.ResponseBody))*/
        logger.debugLog(this.javaClass.getName(), "Finished")
        /*
        api.logging().logToOutput(api.persistence().preferences().getString("com.nickcoblentz.montoya.explorepreferences.keyname"));
        api.persistence().preferences().stringKeys().forEach(key ->{
            api.logging().logToOutput(key);
        });

 */
    }
}